Minecraft client no longer works on NixOS

I used to play Minecraft on NixOS just fine, but ever since the game switched from Mojang IDs to Microsoft IDs, the Linux client no longer works. If I attempt to log in with Mojang, the software says that I am not online, which means that the server it's trying to talk to has disappeared I guess. If I attempt to log in with the Microsoft ID, it does work initially, but then the game won't start saying it experienced an unexpected crash.

![error](https://user-images.githubusercontent.com/28323/175926012-98cc26de-dc4d-4332-a437-61a9a75f0c25.jpg)

Alr so i turned on logs for Java Edition

and Here are The Results:

Vanilla: `12:02:08.002
[STDERR]: 	at enn.a(SourceFile:2424)
12:02:08.002
[STDERR]: 	at enn.a(SourceFile:2419)
12:02:08.002
[STDERR]: 	at net.minecraft.client.main.Main.main(SourceFile:220)
12:02:08.002
[STDERR]: Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.UnsatisfiedLinkError: Failed to locate library: liblwjgl.so [in thread "Render thread"]
12:02:08.003
[STDERR]: 	at org.lwjgl.system.Library.loadSystem(Library.java:164)
12:02:08.003
[STDERR]: 	at org.lwjgl.system.Library.loadSystem(Library.java:63)
12:02:08.003
[STDERR]: 	at org.lwjgl.system.Library.<clinit>(Library.java:51)
12:02:08.003
[STDERR]: 	at org.lwjgl.system.MemoryUtil.<clinit>(MemoryUtil.java:100)
12:02:08.003
[STDERR]: 	at ehh.<clinit>(SourceFile:8)
12:02:08.003
[STDERR]: 	at eie.<init>(SourceFile:48)
12:02:08.003
[STDERR]: 	at eil.<init>(SourceFile:19)
12:02:08.003
[STDERR]: 	at eil.<init>(SourceFile:23)
12:02:08.003
[STDERR]: 	at eil.<clinit>(SourceFile:11)
12:02:08.003
[STDERR]: 	at com.mojang.blaze3d.systems.RenderSystem.<clinit>(SourceFile:50)
12:02:08.003
[STDERR]: 	at net.minecraft.client.main.Main.main(SourceFile:208)
12:02:08.063

12:02:08.063
Process crashed with exit code 1`

Fabric: `12:31:02.409
Preparing to launch minecraft client for fabric-loader-0.14.21-1.20.1
12:31:05.204
Checking installations.
12:31:05.208
Minecraft client fabric-loader-0.14.21-1.20.1 is ready to start.
12:31:05.208
Starting!
12:31:05.209
Using default game log configuration client-1.12.xml (outputs XML)
12:31:06.663
Class path entries reference missing files: /home/ion/.minecraft/libraries/io/netty/netty-transport-native-epoll/4.1.82.Final, linux-aarch_64/netty-transport-native-epoll-4.1.82.Final, linux-aarch_64.jar, /home/ion/.minecraft/libraries/io/netty/netty-transport-native-epoll/4.1.82.Final, linux-x86_64/netty-transport-native-epoll-4.1.82.Final, linux-x86_64.jar, natives-linux/lwjgl-glfw-3.3.1, natives-linux.jar, natives-linux/lwjgl-jemalloc-3.3.1, natives-linux.jar, natives-linux/lwjgl-openal-3.3.1, natives-linux.jar, natives-linux/lwjgl-opengl-3.3.1, natives-linux.jar, natives-linux/lwjgl-stb-3.3.1, natives-linux.jar, natives-linux/lwjgl-tinyfd-3.3.1, natives-linux.jar, natives-linux/lwjgl-3.3.1, natives-linux.jar - the game may not load properly!
12:31:06.665
Loading Minecraft 1.20.1 with Fabric Loader 0.14.21
12:31:06.808
Loading 3 mods:
	- fabricloader 0.14.21
	- java 19
	- minecraft 1.20.1
12:31:06.839
SpongePowered MIXIN Subsystem Version=0.8.5 Source=file:/home/ion/.minecraft/libraries/net/fabricmc/sponge-mixin/0.12.5+mixin.0.8.5/sponge-mixin-0.12.5+mixin.0.8.5.jar Service=Knot/Fabric Env=CLIENT
12:31:10.501
188 Datafixer optimizations took 109 milliseconds`

not sure if this helps but i thought it was a good idea yo put this in

--- a/pkgs/games/minecraft/default.nix
+++ b/pkgs/games/minecraft/default.nix
@@ -151,6 +151,9 @@ stdenv.mkDerivation rec {
     sourceProvenance = with sourceTypes; [ binaryNativeCode ];
     license = licenses.unfree;
     platforms = [ "x86_64-linux" ];
+    # "minecraft-launcher will fail on NixOS for minecraft versions >1.19
+    # try prismlauncher or atlauncher instead"
+    broken = true;
   };
 
   passthru = {

